sdcm/nemesis.py

```diff
@@ -5008,6 +5011,116 @@ def disrupt_disable_binary_gossip_execute_major_compaction(self):
+                        WITH replication = {{'class': 'NetworkTopologyStrategy', \
+                            'replication_factor' : {self.tester.params.get('n_db_nodes')} }};")
+                session.execute(
+                    f"CREATE TABLE IF NOT EXISTS {keyspace_name}.{table_name} (id BIGINT PRIMARY KEY, name text);")
+
+        simulate_node_unavailability = node_operations.block_scylla_ports if use_iptables else node_operations.stop_scylla_with_sigstop
+        working_node = node_operations.get_random_free_verification_node(self.target_node)
+        target_host_id = self.target_node.host_id
+
+        prepare_test_keyspace(self.target_node)
+        try:
+            with simulate_node_unavailability(self.target_node):
+                # target node stopped by Contextmanger. Wait while its status will be updated
+                wait_for(node_operations.is_node_seen_as_down, timeout=600, throw_exc=True,
+                         down_node=self.target_node)
+                self.log.debug("Remove node %s : hostid: %s with stopped scylla from cluster",
+                               self.target_node.name, target_host_id)
+                working_node.run_nodetool(f"removenode {target_host_id}", retry=3)
+                self.log.debug("Wait while all node update topology status")
+                wait_for(node_operations.is_node_removed_from_cluster, timeout=300, throw_exc=True,
+                         node=self.target_node)
+                # Run read barrier on working node to validate that it update raft topology state
+                self.cluster.call_read_barrier(working_node)
+
+            # Context manager at exit  start scylla on target node.
+            # But node already removed from cluster. So any operations from it
+            # should be banned. If query executed succesfull, raise an error
+            assert self.target_node.is_port_used(
+                port=9042, service_name="scylla-service"), f"Port 9042 is closed on node {self.target_node.name}"
+
+            with self.cluster.cql_connection_exclusive(node=self.target_node) as session:
+                try:
+                    stmt = SimpleStatement(f"INSERT INTO {keyspace_name}.{table_name} (id, name) VALUES (1, 'name1');",
+                                           consistency_level=ConsistencyLevel.ALL)
+                    session.execute(stmt)
+                    self.log.error("Banned query passed to cluster from banned node")
+                    raise BannedQueryExecUnexpectedSuccess(
+                        "Query from banned node was executed succesful with Consistency.ALL")
+                except (NoHostAvailable, OperationTimedOut, Unavailable) as exc:
+                    self.log.debug("Query failed with error: %s as expected", exc)
+
+        except Exception as exc:  # pylint: disable=broad-except
+            self.log.error("Isolate node with %s failed with error %s", simulate_node_unavailability.__name__, exc)
```

Comment:
should also `session.execute(f"DROP KEYSPACE IF EXISTS {keyspace_name}")` in case of reraising exception - as rest of code won't be executed and keyspace not cleaned up.

Reply:
It will be deleted in finally block anyway.